tests/extensions/provisioning/verify.go

```diff
@@ -0,0 +1,147 @@
+	rke1                     = "rke1"
+	rke2                     = "rke2"
+	k3s                      = "k3s"
+
+	active                        = "active"
+	ProvisioningSteveResourceType = "provisioning.cattle.io.cluster"
+	provisioning                  = "provisioning"
+	kubernetesUpgrade             = "kubernetes-upgrade"
+	scaleUp                       = "scale-up"
+	scaleDown                     = "scale-down"
+)
+
+// VerifyCluster validates that a downstream cluster and its resources are in a good state, matching a given config.
+func VerifyCluster(t *testing.T, client *rancher.Client, clusterName string, terraformConfig *config.TerraformConfig, terraformOptions *terraform.Options, clusterConfig *config.TerratestConfig) {
+	client, err := client.ReLogin()
+	require.NoError(t, err)
+
+	adminClient, err := rancher.NewClient(client.RancherConfig.AdminToken, client.Session)
+	require.NoError(t, err)
+
+	var expectedKubernetesVersion string
+	module := terraformConfig.Module
+	expectedKubernetesVersion = checkExpectedKubernetesVersion(clusterConfig, expectedKubernetesVersion, module)
+
+	clusterID, err := clusters.GetClusterIDByName(adminClient, clusterName)
+	require.NoError(t, err)
+
+	waitState.IsActiveCluster(adminClient, clusterID)
+	waitState.AreNodesActive(adminClient, clusterID)
+
+	cluster, err := adminClient.Management.Cluster.ByID(clusterID)
+	require.NoError(t, err)
+
+	// EKS is formatted this way due to EKS formatting Kubernetes versions with a random string of letters after the version.
+	if module == eks {
+		assert.Equal(t, expectedKubernetesVersion, cluster.Version.GitVersion[1:5])
+	} else {
+		assert.Equal(t, expectedKubernetesVersion, cluster.Version.GitVersion)
+	}
+
+	clusterToken, err := clusters.CheckServiceAccountTokenSecret(adminClient, cluster.Name)
+	require.NoError(t, err)
+	assert.NotEmpty(t, clusterToken)
+
+	if clusterConfig.PSACT == string(config.RancherPrivileged) || clusterConfig.PSACT == string(config.RancherRestricted) {
+		require.NotEmpty(t, cluster.DefaultPodSecurityAdmissionConfigurationTemplateName)
+
+		_, err := psadeploy.CreateNginxDeployment(adminClient, clusterID, clusterConfig.PSACT)
+		require.NoError(t, err)
+	}
+}
+
+// VerifyUpgradedKubernetesVersion validates that the cluster has the expected upgraded Kubernetes version.
+func VerifyUpgradedKubernetesVersion(t *testing.T, client *rancher.Client, terraformConfig *config.TerraformConfig, clusterName, kubernetesVersion string) {
+	client, err := client.ReLogin()
+	require.NoError(t, err)
+
+	adminClient, err := rancher.NewClient(client.RancherConfig.AdminToken, client.Session)
+	require.NoError(t, err)
+
+	clusterID, err := clusters.GetClusterIDByName(adminClient, clusterName)
+	require.NoError(t, err)
+
+	cluster, err := adminClient.Management.Cluster.ByID(clusterID)
+	require.NoError(t, err)
+
+	if cluster.Version.GitVersion == kubernetesVersion {
+		logrus.Infof("Successfully upgraded cluster to %v!", kubernetesVersion)
+	} else {
```

Comment:
this if/else should be replaced by a require statement intead of logging. Otherwise, the test will have false positives.

Reply:
Resolved